### DISCUSSION TITLE
Delete a message at the beginning

### DISCUSSION THREAD

**Discussion by sr-26** [bbc/sqs-consumer#441] [Help] [answered]
### Summary

Hello! Thank you for developing such an impressive library. I'd like to confirm whether there's a method to delete a message at the beginning of the handler. I've discovered one approach, which involves setting "shouldDeleteMessages" to false and creating a custom delete handler. Are there any alternative methods supported by this library that I may not be aware of ?

**Comment by nicholasgriffintn** [comment, marked as answer]
Hey, I'm not really sure about the use case here, but messages are only deleted from the queue by Consumer once the handler function has completed successfully. I don't think you'd want to delete at the beginning, at least I can't think of a use case for that.

With Consumer, we're really only looking to provide the basic one size fits all config as well, it isn't purposed for super complex things.
